ipatests/test_integration/test_rbcd.py

```diff
@@ -0,0 +1,105 @@
+#
+# Copyright (C) 2023  FreeIPA Contributors see COPYING for license
+#
+
+"""This module provides tests for resource-based constrained delegation (RBCD)
+   - check scenario between ipa_client and AD (use S4U2PRoxy)
+    (FreeIPA currently does not have support for IPA to IPA trust and does not
+    provide a working two-way trust with Active Directory)
+"""
+
+from __future__ import absolute_import
+
+from functools import partial
+import textwrap
+import re
+import os
+import pytest
+
+from ipatests.test_integration.base import IntegrationTest
+from ipatests.pytest_ipa.integration import tasks
+
+class TestRBCD(IntegrationTest):
+    topology = 'star'
+    num_clients = 1
+    num_ad_domains = 1
+
+    ipa_user1 = 'alice'
+    ipa_user1_password = 'SecretUser123'
+    ad_user_login = 'ad_testuser'
+    ad_user_password = 'Secret123'
+    ipa_test_group = 'ipa_testgroup'
+    ad_test_group = 'testgroup'
+
+    @classmethod
+    def install(cls, mh):
+        if cls.domain_level is not None:
+            domain_level = cls.domain_level
+        else:
+            domain_level = cls.master.config.domain_level
+        tasks.install_topo(cls.topology,
+                           cls.master, cls.replicas,
+                           cls.clients, domain_level,
+                           clients_extra_args=('--mkhomedir',))
+
+        cls.ad = cls.ads[0]
+        cls.ipa-client = cls.clients[0]
+        cls.ad_user = '{}@{}'.format(cls.ad_user_login, cls.ad.domain.name)
+
+        tasks.install_adtrust(cls.master)
+        tasks.configure_dns_for_trust(cls.master, cls.ad)
+        tasks.configure_windows_dns_for_trust(cls.ad, cls.master)
+        tasks.establish_trust_with_ad(cls.master, cls.ad.domain.name,
+                                      extra_args=['--two-way=false'])
```

Comment:
You must use `--two-way=true` for any cross-realm S4U scenarios. Please see https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-sfu/1fb9caca-449f-4183-8f7a-1a5fc7e7290a and https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-sfu/f35b6902-6f5e-4cd0-be64-c50bbaaf54a5 for more details.